.github/workflows/link-check.yml

```diff
@@ -0,0 +1,48 @@
+name: Build and check Links
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  build_and_serve:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Install Dependencies
+        run: yarn install
```

Comment:
Use `yarn install --frozen-lockfile` instead.

You can check out this npm blog: [Introducing npm ci for faster, more reliable builds](https://blog.npmjs.org/post/171556855892/introducing-npm-ci-for-faster-more-reliable).

`yarn install --frozen-lockfile` does the same thing as `npm ci`. Since we are using Yarn, we are good to go.